src/wallet/wallet.cpp

```diff
@@ -1762,6 +1772,27 @@ std::optional<libzcash::AccountId> CWallet::FindAccountForSelector(const ZTXOSel
                 result = self->GetUnifiedAccountId(ufvkid.value());
             }
         },
+        [&](const libzcash::UnifiedAddress& addr) {
+            std::optional<libzcash::AccountId> singleAccount;
+            for (const auto& receiver : addr) {
+                auto meta = GetUFVKMetadataForReceiver(receiver);
+                if (meta.has_value()) {
+                    auto tmp = self->GetUnifiedAccountId(meta.value().first);
+                    if (singleAccount.has_value()) {
+                        // If a UA corresponds to more than one account, we cannot return
+                        // a sensible result, so we report it as corresponding to no
+                        // account.
+                        if (tmp.has_value() && singleAccount.value() != tmp.value()) {
+                            singleAccount = std::nullopt;
+                            break;
+                        }
+                    } else {
+                        singleAccount = tmp;
+                    }
+                }
+            }
+            result = singleAccount;
+        },
```

Comment:
Would it be equivalent to use `GetUFVKMetadataForAddress(addr)`?

```suggestion
        [&](const libzcash::UnifiedAddress& addr) {
            auto meta = GetUFVKMetadataForAddress(addr);
            if (meta.has_value()) {
                result = meta.value().first;
            }
```

Reply:
`meta.value().first` is the ufvkid, not the account id, but this is otherwise a good idea.

Reply:
Updated my suggestion.